app/messages/buildpack_upload_message.rb

```diff
@@ -53,7 +54,10 @@ def is_archive
 
       return if mime_bits =~ /^#{VCAP::CloudController::GZIP_MIME}/ || mime_bits =~ /^#{VCAP::CloudController::ZIP_MIME}/
 
-      errors.add(:base, "#{bits_name} is not a zip or gzip archive")
+      mime_bits_at_offset = File.read(bits_path, 8, 257)
+      return if mime_bits_at_offset =~  /^#{VCAP::CloudController::CNB_MIME}/
+
+      errors.add(:base, "#{bits_name} is not a zip or gzip archive or cnb file")
```

Comment:
CNB file has to specific format, it's simly a tarball

Reply:
the cnb file is part of the buildpack spec (https://github.com/buildpacks/spec/blob/main/distribution.md#buildpackage). Even though its just a tar, we want to imply that we are just following the cnb spec and if that format changes in the future we would expect to adapt to that change.